src/main/java/jsr223/kubernetes/KubernetesScriptEngine.java

```diff
@@ -364,7 +364,31 @@ public Object eval(Reader reader, ScriptContext context) throws ScriptException
             log.debug("Failed to convert Reader into StringWriter. Not possible to execute Kubernetes task.", e);
         }
 
-        return eval(stringWriter.toString(), context);
+        // Needed to guarantee cleanup in case of kill
+        Thread shutdownHook = null;
+
+        //Add a shutdownHook to remove the kubernetes pod in case of kill
+        shutdownHook = new Thread() {
+            @Override
+            public void run() {
+                cleanKubernetesResources();
+                deleteKubernetesManifestFile();
+            }
+        };
+        Runtime.getRuntime().addShutdownHook(shutdownHook);
+        Object return_val = null;
+
+        try {
+            return_val = eval(stringWriter.toString(), context);
+        } catch (Exception e) {
+
```

Comment:
you should add the shutdown hook inside the eval(String, context) method.

Otherwise the change is applied only when the script engine is used with a Reader